sycl/include/sycl/types.hpp

```diff
@@ -905,11 +905,42 @@ template <typename Type, int NumElements> class vec {
   // Implement operator [] in the same way for host and device.
   // TODO: change host side implementation when underlying type for host side
   // will be changed to std::array.
-  const DataT &operator[](int i) const {
+  // NOTE: aliasing on bfloat16 may lead to problems if aggressively optimized.
+  // specializing with noinline to avoid as workaround.
+
+  template <typename T = DataT>
+  typename std::enable_if<!std::is_same<T, sycl::ext::oneapi::bfloat16>::value,
+                          const DataT &>::type
```

Comment:
```suggestion
  typename std::enable_if_t<!std::is_same_v<T, sycl::ext::oneapi::bfloat16>,
                          const DataT &>
```
Please, apply below too.